src/python/turicreate/toolkits/one_shot_object_detector/util/_augmentation.py

```diff
@@ -39,6 +40,8 @@ def preview_synthetic_training_data(data,
     out : SFrame
         An SFrame of sythetically generated annotated training data.
     """
+    if (isinstance(backgrounds, _tc.SArray)  and len(backgrounds) == 0):
```

Comment:
For tidiness, could we move this into `_error_handling.py`? Perhaps we just need to add `backgrounds` to the argument list for `check_one_shot_input`